doc/specs/#4999 - Improved keyboard handling in Conpty.md

```diff
@@ -0,0 +1,500 @@
+  have the same fidelity that Win32 mouse events do.
+  - We could enable this with a different terminating character, to identify
+    which type of `INPUT_RECORD` event we're encoding.
+* Client applications that want to be able to read full Win32 keyboard input
+  from `conhost` _using VT_ will also be able to use `^[[?9001h` to do this. If
+  they emit `^[[?9001h`, then conhost will switch itself into
+  `win32-input-mode`, and the client will read `win32-input-mode` encoded
+  sequences as input. This could enable other cross-platform applications to
+  also use win32-like input in the future.
+
+## Options Considered
+
+### Create our own format for `INPUT_RECORD`s
+
+* If we wanted to do this, then we'd probably want to have the Terminal only
+  send input as this format, and not use the existing translator to synthesize
+  VT sequences
+    - Consider sending a ctrl down, '^A', ctrl up. We wouldn't want to send this
+      as three sequences, because conpty will take the '^A' and synthesize
+      _another_ ctrl down, ctrl up pair.
+* With conpty passthrough mode, we'd still need the `InputStateMachineEngine`
+  to convert these sequences into INPUT_RECORDs to translate back to VT
+* Wouldn't really expect client apps to ever _need_ this format, but it could
+  always be possible for them to need it in the future.
+
+#### Pros:
+* Definitely gets us all the information that we need.
+* Can handle solo modifiers
+* Can handle keydown and keyup separately
+* We can make the sequence however we want to parse it.
+
+#### Cons:
+* No reference implementation, so we'd be flying blind
+* We'd be defining our own VT sequences for these, which we've never done
+  before. This was _inevitable_, however, this is still the first time we'd be
+  doing this.
+* Something about Microsoft extending well-defined protocols being bad 😆
+* By having the Terminal send all input as _this protocol_, VT Input passthrough
+  to apps that want VT input won't work anymore for the Terminal. That's _okay_
+
+### kitty extension
+[Reference](https://sw.kovidgoyal.net/kitty/protocol-extensions.html#keyboard-handling)
+
+#### Pros:
+* Not terribly difficult to decode
+* Unique from anything else we'd be processing, as it's an APC sequence
+  (`\x1b_`)
+* From their docs: > All printable key presses without modifier keys are sent
+  just as in the normal mode. ... For non printable keys and key combinations
+  including one or more modifiers, an escape sequence encoding the key event is
+  sent
+   - I think like this. ASCII and other keyboard layout chars (things that would
+     hit `SendChar`) would still just come through as the normal char.
+
+#### Cons:
+* Their encoding table is _mad_. [Look at
+  this](https://sw.kovidgoyal.net/kitty/key-encoding.html). What order is that
+  in? Obviously the first column is sorted alphabetically, but the mapping of
+  key->char is in a seemingly bonkers order.

```

Comment:
Might be a bit rude to call it "mad" and "bonkers".

Reply:
Yea that's verbatim from my notes. I'll polish.